components/camel-azure-eventhubs/src/main/java/org/apache/camel/component/azure/eventhubs/operations/EventHubsProducerOperations.java

```diff
@@ -0,0 +1,113 @@
+import org.apache.camel.util.ObjectHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+
+public class EventHubsProducerOperations {
+
+    private static final Logger LOG = LoggerFactory.getLogger(EventHubsProducerOperations.class);
+
+    private final EventHubProducerAsyncClient producerAsyncClient;
+    private final EventHubsConfigurationOptionsProxy configurationOptionsProxy;
+
+    public EventHubsProducerOperations(final EventHubProducerAsyncClient producerAsyncClient, final EventHubsConfiguration configuration) {
+        ObjectHelper.notNull(producerAsyncClient, "client cannot be null");
+
+        this.producerAsyncClient = producerAsyncClient;
+        configurationOptionsProxy = new EventHubsConfigurationOptionsProxy(configuration);
+    }
+
+    public boolean sendEvents(final Exchange exchange, final AsyncCallback callback) {
+        ObjectHelper.notNull(exchange, "exchange cannot be null");
+        ObjectHelper.notNull(callback, "callback cannot be null");
+
+        final SendOptions sendOptions = createSendOptions(configurationOptionsProxy.getPartitionKey(exchange), configurationOptionsProxy.getPartitionId(exchange));
+        final Iterable<EventData> eventData = createEventData(exchange);
+
+        return sendAsyncEvents(eventData, sendOptions, exchange, callback);
+    }
+
+    private boolean sendAsyncEvents(final Iterable<EventData> eventData, final SendOptions sendOptions, final Exchange exchange, final AsyncCallback asyncCallback) {
+        sendAsyncEventsWithSuitableMethod(eventData, sendOptions, asyncCallback)
+                .subscribe(unused -> LOG.debug("Processed one event..."), error -> {
+                    // error but we continue
+                    LOG.debug("Error processing async exchange with error:" + error.getMessage());
+                    exchange.setException(error);
+                }, () -> {
+                    // we are done from everything, so mark it as sync done
+                    LOG.debug("All events with exchange have been sent successfully.");
+                });
+
+        return false;
+    }
+
+    private Mono<Void> sendAsyncEventsWithSuitableMethod(final Iterable<EventData> eventData, final SendOptions sendOptions,
+                                                         final AsyncCallback asyncCallback) {
+        final Mono<Void> deferredSendResult;
+
+        if (ObjectHelper.isEmpty(sendOptions)) {
+            deferredSendResult = producerAsyncClient.send(eventData);
+        } else {
+            deferredSendResult = producerAsyncClient.send(eventData, sendOptions);
+        }
+
+        // call callback on false since this process is running on the scheduler method of reactor and not needed
+        // to have callback to returns true. Also false, will tell camel async to watch method to which it returns
+        asyncCallback.done(false);
```

Comment:
I would asumme this asyncCallback.done(false) should be in the previous method inside those 2 callbacks after ( // error but we continue) and also after ( // we are done from everything, so mark it as sync done). As this Camel callback should only be invoked when the azure client is fully complete with its work (either in error or with success).